backend/oqtopus_cloud/provider/routers/jobs.py

```diff
@@ -218,32 +251,51 @@ def decode_job_status(s: str) -> JobStatus | ValueError:
         return ValueError(f"Failed to decode JobStatus: {str(err)}")
 
 
-def model_to_schema(model: Job) -> JobDef | ValueError:
+def model_to_schema(
+    model: Union[Job, Row],
+) -> Union[JobDef, GetJobsResponse] | ValueError:
     def decode_job_info(j: Any) -> JobInfo | ValueError:
         try:
             jobinfo = JobInfo.model_validate(j)
             return jobinfo
         except Exception as e:
             return ValueError(f"Failed to decode job_info: {str(e)}")
 
-    status = decode_job_status(model.status)
-    if isinstance(status, ValueError):
-        return status
-
-    job_info = decode_job_info(json.loads(model.job_info))
-    if isinstance(job_info, ValueError):
-        return job_info
-    return JobDef(
-        job_id=model.id,
-        name=model.name,
-        description=model.description,
-        device_id=model.device_id,
-        shots=model.shots,
-        job_info=job_info,
-        status=status,
-        transpiler_info=model.transpiler_info,
-        mitigation_info=model.mitigation_info,
-        simulator_info=model.simulator_info,
-        created_at=model.created_at,
-        updated_at=model.updated_at,
-    )
+    if hasattr(model, "status"):
+        status = decode_job_status(model.status)
+    if hasattr(model, "job_info"):
+        job_info = decode_job_info(json.loads(model.job_info))
+
+    if type(model) is Job:
+        if isinstance(job_info, ValueError):
```

Comment:
I recommend early-returning in the case of failure.
Additionally, I think you must do same check for status & job_info in the elif branch.

Reply:
https://github.com/oqtopus-team/oqtopus-cloud/blob/95b0788414394d812526f434c26ecd95285cd793/backend/oqtopus_cloud/provider/routers/jobs.py#L299-L310

add exception handling for status & job_info in case with fields parameter is given